process/smartContract/hooks/blockChainHook.go

```diff
@@ -771,6 +775,24 @@ func (bh *BlockChainHookImpl) EpochConfirmed(epoch uint32, _ uint64) {
 	}
 }
 
+// ExecuteSmartContractCallOnOtherVM on another VM
+func (bh *BlockChainHookImpl) ExecuteSmartContractCallOnOtherVM(input *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
+	vmExec, _, err := scrCommon.FindVMByScAddress(bh.vmContainer, input.RecipientAddr)
+	if err != nil {
+		return nil, err
+	}
+	return vmExec.RunSmartContractCall(input)
```

Comment:
I know this can be the case only when we have a bad VM setup (but we can not always control bad setups being made - especially from other contributors) but I think we should prevent the execution of calls *on the same VM on which the blockchain hook was set on* as that can trigger some very wrong outputs.

Reply:
This is a check done in VM before the call is made. I see that blockchain hook has only a reference to a vm container not a certain vm. @sasurobert ?

Reply:
This has to be a check on the VM.